include/envoy/http/access_log.h

```diff
@@ -89,9 +91,10 @@ class RequestInfo {
   virtual std::chrono::milliseconds duration() const PURE;
 
   /**
-   * @return the failure reason for richer log experience.
+   * @return the failure reason for richer log experience, this can be represented as binary OR of
+   * FailureReason enum values.
    */
-  virtual FailureReason failureReason() const PURE;
+  virtual uint64_t failureReason() const PURE;
```

Comment:
Yeah I guess so, which makes it a larger change, but I would probably do the right thing here. I'm open to other name ideas also.